python/src/aiconfig/default_parsers/openai.py

```diff
@@ -349,3 +349,45 @@ def refine_chat_completion_params(model_settings):
             completion_data[key] = model_settings[key]
 
     return completion_data
+
+
+def add_prompt_as_message(
+    prompt: Prompt, aiconfig: "AIConfigRuntime", messages: list, params=None
+):
+    """
+    Converts a given prompt to a message and adds it to the specified messages list.
+
+    Note:
+    - If the prompt contains valid input, it's treated as a user message.
+    - If the prompt has a custom role, function call, or name, these attributes are included in the message.
+    - If an AI model output exists, it is appended to the messages list.
+    """
+    if has_valid_prompt_string(prompt):
+        resolved_prompt = resolve_prompt(prompt, params, aiconfig)
+        messages.append({"content": resolved_prompt, "role": "user"})
+    else:
+        # prompt data can have function role
+        prompt_input = prompt.input
+        role = prompt_input.role if hasattr(prompt_input, "role") else "user"
+        fn_call = prompt_input.function_call if hasattr(prompt_input, "function_call") else None
+        name = prompt_input.name if hasattr(prompt_input, "name") else None
+        messages.append(
+            {"content": resolved_prompt, "role": role, "function_call": fn_call, "name": name}
+        )
+
+    output = aiconfig.get_latest_output(prompt)
+    if output:
+        if output.output_type == "execute_result":
+            output_message = output.data
+            if output_message["role"] == "assistant":
+                messages.append(output_message)
+    return messages
+
+
+def has_valid_prompt_string(prompt: Prompt):
```

Comment:
this feels a bit brittle. Can a prompt have a statically known structure?

Reply:
No, The type definition of the prompt input is either a string or arbitrary data. We would have dynamically "parse" it.

Any suggestions on how you would make this not as brittle?

Reply:
Need some more context - why can it be arbitrary data? What are the use cases?

Reply:
Let's chat in person on this. For more context, this class is a parser for OpenAI models specifically. It serializes and deserializes prompts in a specific structure in aiconfig, and so this should be safe to do in this class because only prompts that were serialized by this class are processed by it.